### PR TITLE
Next.js Image Component: Width/Height Update

### PR DESCRIPTION
**Pull request by KeazoN**
This PR addresses an issue with the `Next.js Image component` where the `width `and `height `attributes were marked as optional but caused errors when omitted. These attributes are now required to ensure consistent behavior and prevent runtime errors.

This update improves the developer experience and ensures reliable implementation of the Image component.

- Resolved an issue where optional width and height attributes caused errors when omitted.
- Updated the implementation to enforce these attributes as required for consistency and reliability.

### CODE REVIEW

**Comment by sannajammeh** [comment]
What about fill mode?

File: packages/next/src/shared/lib/get-img-props.ts [hunk 1]
```diff
@@ -30,8 +30,8 @@ export type ImageProps = Omit<
 > & {
   src: string | StaticImport
   alt: string
-  width?: number | `${number}`
-  height?: number | `${number}`
+  width: number | `${number}`
+  height: number | `${number}`
```
Comment (youneshenniwrites) [inline packages/next/src/shared/lib/get-img-props.ts:34]: How about the the rest of the props in this file such as `fill` and `loader`? Please explain why only width and height are causing issues when declared as optional.

**Review by youneshenniwrites — CHANGES_REQUESTED** [review]

File: packages/next/src/shared/lib/get-img-props.ts (hunk 1 above)
Comment (KeazoN) [reply packages/next/src/shared/lib/get-img-props.ts]: - Made `width` and `height` mandatory when `fill` is not set or is `false`.
- Allowed `width` and `height` to be optional when `fill` is `true`.
- Improved type safety for ImageProps to ensure consistent usage in components.

File: packages/next/src/shared/lib/get-img-props.ts [hunk 2]
```diff
@@ -67,7 +67,10 @@ export type ImageProps = Omit<
    * @deprecated This prop does not do anything.
    */
   lazyRoot?: string
-}
+} & (
```
Comment (youneshenniwrites) [inline packages/next/src/shared/lib/get-img-props.ts:70]: Very clever pattern! 👏 The only suggestion I have is to add a brief comment for clarity, as this logic might be a bit hard to follow for some developers. For example:

```ts
/**
 * This logic ensures:
 * - When `fill` is `true`, `width` and `height` are optional and can be either numbers or numeric strings.
 * - When `fill` is `false` (or omitted), `width` and `height` are required and follow the same type rules.
 */
```

**Review by youneshenniwrites — CHANGES_REQUESTED** [review]

File: packages/next/src/shared/lib/get-img-props.ts (hunk 2 above)
Comment (KeazoN) [reply packages/next/src/shared/lib/get-img-props.ts]: Thanks 🙌 Of course I added and committed

**Review by youneshenniwrites — APPROVED** [review]

File: packages/next/src/shared/lib/get-img-props.ts (hunk 2 above)
Comment (youneshenniwrites) [reply packages/next/src/shared/lib/get-img-props.ts]: Well done :)

**Review by youneshenniwrites — APPROVED** [review]
LGTM 👍🏼

**Comment by KeazoN** [comment]
Isn't anyone interested?

**Review by KeazoN — COMMENTED** [review]
LGTM
